### DISCUSSION TITLE
I want icons on MY desktop !!!

### DISCUSSION THREAD

**Discussion by zachari043** [elementary/triage#257] [Q&A] [answered]
I think it's a well know approved feature of a desktop, It's cool for developers to explore new point of view, but not there... It's your app your choices, but at the second it's installed it become "My" computer my desktop, I think you must add a setting option like "Show files on desktop" set to no by default if you wish but as many users, I want my desktop icons ! please !

**Comment by 1280px** [comment, marked as answer]
The icons were removed a long time ago, so chances are they're not ever coming back :c

You can try my script: https://github.com/1280px/eOS-xfdesktop4
It's still WIP, but basically it installs xfdesktop4 with a few tweaks to look better on Pantheon, and Win+D shortcut to show/hide desktop (works a little weird now, I'll probably rewrite it to work exactly like on Windows).

**Reply by danirabbit** [reply]
There was never anything in Pantheon to draw icons on the wallpaper, but I agree the best way to get this functionality is to install a 3rd party program like XFDesktop4 or there was [Desktop Folder](https://github.com/spheras/desktopfolder) which was made for elementary OS

**Reply by zachari043** [reply]
Desktop folders no longer available in the store...  "There was never anything in Pantheon to draw icons on the wallpaper"  and what? is not Pantheon a part of elementary ? add it to the source code...  Pantheon the only _desktop_ environment without a desktop !

**Reply by zachari043** [reply]
Is this a design choice ?

**Reply by 1280px** [reply]
I actually used to use DF, but it seems to be quite buggy (i.e. if you mass downloading stuff right to your desktop, most of the files won't appear, and there is no way to update it), so it's better to go with xfdesktop if you only want the icons
